lib/routes.json

```diff
@@ -1663,7 +1710,34 @@
         },
         "tree": {
           "required": true,
-          "type": "object"
+          "type": "object[]"
+        },
+        "tree.content": {
+          "type": "string"
+        },
+        "tree.mode": {
+          "enum": [
+            "100644",
+            "100755",
+            "040000",
+            "160000",
+            "120000"
+          ],
+          "type": "string"
+        },
+        "tree.path": {
+          "type": "string"
+        },
+        "tree.sha": {
+          "type": "string"
+        },
+        "tree.type": {
+          "enum": [
+            "blob",
+            "tree",
+            "commit"
+          ],
+          "type": "string"
```

Comment:
This is correct, but tricky. `tree` is an array of objects, not sure if `tree.content` etc is the right notation, or if we should use `tree[].content` or something else.